src/main/java/com/cloudlabs/server/snapshot/SnapshotController.java

```diff
@@ -0,0 +1,85 @@
+package com.cloudlabs.server.snapshot;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.cloudlabs.server.compute.dto.ComputeDTO;
+import com.cloudlabs.server.snapshot.dto.SaveSnapshotDto;
+
+@CrossOrigin(origins = {"${app.security.cors.origin}"})
+@RequestMapping("snapshot")
+@RestController()
+public class SnapshotController {
+    
+    private SnapshotService snapshotService;
+
+    //private SnapshotDto snapshotDto;
+
+    public SnapshotController(SnapshotService snapshotService) {
+        this.snapshotService = snapshotService;
+    }
+
+    @PostMapping(path = "create", consumes = (MediaType.APPLICATION_FORM_URLENCODED_VALUE))
```

Comment:
If possible, consider using JSON, instead of application form since we are following the REST APi specifications. Follow the project convention, and have a consistent pattern.

This applies for the rest of the methods below as well, that involves accepting any other format rather than JSON.